### DISCUSSION TITLE
Text alignment for textbox doesn't work the way its assumed to

### DISCUSSION THREAD

**Discussion by Passeriform** [davatorium/rofi#1636] [(Possible) Issues] [answered]
I have the following config for a button-type textbox:
```
#button-toggle-discoverable {
	content: "Toggle Discoverable";
	vertical-align: 0.5;
	horizontal-align: 0.5;
	cursor: pointer;
	action: "kb-custom-3";
}
```

If I end up having content string that is larger than the resulting button texbox's width it wraps to the next line, ignoring the `horizontal-align` attribute and making itself left aligned.

I think its actually aligning the button fine. Just that the text inside it is not aligned.

I initially thought rofi allowed alignment of text directly instead of textbox/button widget. If that's not implemented/isn't a bug, I'd love to have a feature for this.

**Comment by DaveDavenport** [comment, marked as answer]
The horizontal-align is done on the bounding box that the font renderer (pango) returns. I don't actually tell it to center its content.
This would need to be added, either as a separate option, or something like:
horizontal align > 0.8 -> align right
horizontal align < 0.2 -> align left
else -> align center.

**Reply by DaveDavenport** [reply]
this solution will be available in git soon.

**Reply by Passeriform** [reply]
Thanks @DaveDavenport

**Comment by vredesbyyrd** [comment]
On latest master - `rofi -e "This is the end" -no-config -theme-str "textbox {horizontal-align: 0.5;}" ` appears to right align text within the box. Is this expected behavior? Its an honest question and I am wondering if I have overlooked something.

**Reply by DaveDavenport** [reply]
hmm pango is not doing what it sais..  fun.  PANGO_ALIGN_CENTER is set for that example.
Ill debug it.

**Reply by vredesbyyrd** [reply]
pango gonna pango :/

**Reply by DaveDavenport** [reply]
possible fix pushed.  Me and pango have another interpretation of top left corner.

**Reply by vredesbyyrd** [reply]
Nice. That did the trick. Thanks.
